tests/evm/test_sdiv_smod.py

```diff
@@ -0,0 +1,107 @@
+import pytest
+from zkevm_specs.evm import (
+    ExecutionState,
+    StepState,
+    Opcode,
+    verify_steps,
+    Tables,
+    RWDictionary,
+    Block,
+    Bytecode,
+)
+from zkevm_specs.evm.execution.sdiv_smod import get_abs, get_neg, is_neg
+from zkevm_specs.util import rand_fq, rand_word, RLC
+from common import generate_nasty_tests
+
+TESTING_DATA = [
+    (Opcode.SDIV, 0xFFFFFF, 0xABC),
+    (Opcode.SDIV, 0xABC, 0xFFFFFF),
+    (Opcode.SDIV, 0xFFFFFF, 0xFFFFFFF),
+    (Opcode.SDIV, 0xABC, 0),
+    (Opcode.SDIV, (1 << 256) - 1, 0xABCDEF),
+    (Opcode.SDIV, 0xABCDEF, (1 << 256) - 1),
+    (Opcode.SDIV, 1 << 255, (1 << 256) - 1),
+    (Opcode.SMOD, 0xFFFFFF, 0xABC),
+    (Opcode.SMOD, 0xABC, 0xFFFFFF),
+    (Opcode.SMOD, 0xFFFFFF, 0xFFFFFFF),
+    (Opcode.SMOD, 0xABC, 0),
+    (Opcode.SMOD, (1 << 256) - 1, 0xABCDEF),
+    (Opcode.SMOD, 0xABCDEF, (1 << 256) - 1),
+    (Opcode.SMOD, 1 << 255, (1 << 256) - 1),
+    (Opcode.SDIV, rand_word(), rand_word()),
+    (Opcode.SMOD, rand_word(), rand_word()),
+]
+
+generate_nasty_tests(TESTING_DATA, (Opcode.SDIV, Opcode.SMOD))
+
+
+@pytest.mark.parametrize("opcode, a, b", TESTING_DATA)
+def test_sdiv_smod(opcode: Opcode, a: int, b: int):
+    a_abs = get_abs(a)
+    b_abs = get_abs(b)
+    a_is_neg = is_neg(a)
+    b_is_neg = is_neg(b)
+    if opcode == Opcode.SDIV:
+        if b == 0:
+            c = 0
+        elif a_is_neg == b_is_neg:
+            c = a_abs // b_abs
```

Comment:
Since it is the test code and calculates with only the integer value (as `RLC(x).int_value`).

For this special case, `a = 1 << 255` and `b = (1 << 256) - 1` (as signed integer `-1`),
`a_abs = get_abs(a) == 1 << 255` and `b_abs = get_abs(b) == 1`,
so `c = 1 << 255`, then push `RLC(1 << 255, randomness)` to stack.

The value of `RLC(1 << 255)` should be signed integer `-(1 << 255)`. Since a signed RLC could only express an integer of `[-(1 << 255), + 1 << 255)` without `+ 1 << 255`.

This test case is added as [the above code](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/213/files/1da3ed09e575bb2cde04fb9550080078ce385d63#diff-88c8bad5651317741345bb604db8e04971f7d0f9d9754abfdfb75a88ad12a37cR23).